doc/tutorials/tutorial_X8.rst

```diff
@@ -176,6 +178,51 @@ counts:
 >>> counts[(0, 0, 0, 0, 0, 0, 0)]
 2
 
+Job management
+~~~~~~~~~~~~~~
+
+Above, when we called ``eng.run()``, we had to wait for the
+remote device to execute the program and the result to be returned before we could
+continue executing code. That is, ``eng.run()`` is a **blocking** method.
+
+Sometimes, however, it is useful to submit the program
+and continue performing computation locally, every now and again checking to see
+if the job is complete and the results are ready. This is possible with
+the **non-blocking** :meth:`eng.run_async() <.RemoteEngine.run_async>` method:
+
+>>> job = engine.run_async(program, shots=100)
+
+This method is complete as soon as the job has been submitted; no need to wait for
+the job to move through the queue, execute, and for the results to be returned.
+
+Unlike ``eng.run()``, it returns a :class:`~.Job` instance, which allows us to
+check the status of our submitted job:
+
+>>> job.id
+"e6ead866-04c9-4d48-ba28-680e8639fc41"
+>>> job.status
+"queued"
+
+When the job result is ready, it is available via the ``result`` property.
+If the job result is not yet available, however, an ``InvalidJobOperationError``
+will be raised:
+
+>>> job.result
+InvalidJobOperationError
+
+To check when the results are ready, the job can be refreshed, and the status
+checked:
+
+>>> job.refresh()
+>>> job.status
+"complete"
+>>> result = job.result
+>>> result.samples.shape
+(100, 8)
+
+Finally, an incomplete job can be *cancelled* by calling :meth:`job.cancel() <.Job.cancel>`.
+
+
 .. note::
 
     The :class:`~.operation` decorator allows you to create your own Strawberry Fields
```

Comment:
I think this note has got left behind and belongs more naturally in the previous section.
E.g., I'd put it around line 130

Reply:
Have moved it up!